src/map/impls.rs

```diff
@@ -103,8 +114,16 @@ impl<K, V> From<Vec<(K, V)>> for VecMap<K, V>
 where
     K: Eq,
 {
-    fn from(vec: Vec<(K, V)>) -> Self {
-        VecMap::from_iter(vec)
+    /// Constructs map from a vector of `(key → value)` pairs.
+    ///
+    /// **Note**: This conversion has a quadratic complexity because the
+    /// conversion preserve order of elements while at the same time having to
+    /// make sure no duplicate keys exist.  To avoid it, sort and deduplicate
+    /// the vector and use [`VecMap::from_vec_unchecked`] instead.
+    fn from(mut vec: Vec<(K, V)>) -> Self {
+        crate::dedup(&mut vec, |rhs, lhs| rhs.0 == lhs.0);
+        // SUPER: We’ve just deduplicated the elements.
```

Comment:
You probably meant to say `SAFETY` here instead of `SUPER`, I guess?

```suggestion
        // SAFETY: We’ve just deduplicated the elements.
```